File: Codeforces/A_Phone_Desktop.java
import java.util.Scanner;

public class A_Phone_Desktop {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int T = scanner.nextInt(); // total test cases

    for (int i = 0; i < T; i++) {
      int x = scanner.nextInt();
      int y = scanner.nextInt();

      int screensUsed = (int) Math.ceil(y / 2.0); // minimum screens needed for 2x2
      int unusedCells = (screensUsed * 15) - (y * 4); // unused cells after placing 2x2

      if (unusedCells >= x) { // enough cells for 1x1?
        System.out.println(screensUsed);
        continue;
      }

      // otherwise add more screens
      x -= unusedCells; // use `unusedCells`, place remaining 1x1 on new screens
      screensUsed += Math.ceil((double) x / 15);
      System.out.println(screensUsed);
    }
    scanner.close();
  }
}

File: Codeforces/A_Phone_Desktop.py
""" 
Read x and y
    Calculate screens_needed = ceil(y / 2)
    Calculate total_cells_used = y * 4
    Calculate unused_cells = (screens_needed * 15) - total_cells_used
    If unused_cells is enough for 1x1 icons:
        Print screens_needed
    Else:
        Calculate remaining_1x1 = x - unused_cells
        Calculate additional_screens = ceil(remaining_1x1 / 15)
        Print screens_needed + additional_screens
"""

import math

T = int(input())  # total test cases

for _ in range(T):
  # handle each test case here
  x, y = map(int, input().split())

  screens_used = math.ceil(y/2)  # minimum screens needed for 2x2
  unused_cells = (screens_used*15) - (y*4)  # unused cells after placing 2x2

  if unused_cells >= x:  # enough cells for 1x1?
    print(screens_used)
    continue

  # otherwise add more screens
  x -= unused_cells  # use `unused_cells`, place remaining 1x1 on new screens
  screens_used += math.ceil(x/15)
  print(screens_used)
